pkg/virt-handler/vm_test.go

```diff
@@ -78,3 +93,116 @@ var _ = Describe("VM", func() {
 		ctrl.Finish()
 	})
 })
+
+var _ = Describe("PVC", func() {
+	RegisterFailHandler(Fail)
+
+	logging.DefaultLogger().SetIOWriter(GinkgoWriter)
+
+	var (
+		expectedPVC kubev1.PersistentVolumeClaim
+		expectedPV  kubev1.PersistentVolume
+		server      *ghttp.Server
+	)
+
+	BeforeEach(func() {
+		expectedPVC = kubev1.PersistentVolumeClaim{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "PersistentVolumeClaim",
+				APIVersion: "v1",
+			},
+			Spec: kubev1.PersistentVolumeClaimSpec{
+				VolumeName: "disk-01",
+			},
+			Status: kubev1.PersistentVolumeClaimStatus{
+				Phase: kubev1.ClaimBound,
+			},
+		}
+
+		source := kubev1.ISCSIVolumeSource{
+			IQN:          "iqn.2009-02.com.test:for.all",
+			Lun:          1,
+			TargetPortal: "127.0.0.1:6543",
+		}
+
+		expectedPV = kubev1.PersistentVolume{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "PersistentVolume",
+				APIVersion: "v1",
+			},
+			Spec: kubev1.PersistentVolumeSpec{
+				PersistentVolumeSource: kubev1.PersistentVolumeSource{
+					ISCSI: &source,
+				},
+			},
+		}
+
+		server = ghttp.NewServer()
+		server.AppendHandlers(
+			ghttp.CombineHandlers(
+				ghttp.VerifyRequest("GET", "/api/v1/namespaces/default/persistentvolumeclaims/test-claim"),
+				ghttp.RespondWithJSONEncoded(http.StatusOK, expectedPVC),
+			),
+			ghttp.CombineHandlers(
+				ghttp.VerifyRequest("GET", "/api/v1/persistentvolumes/disk-01"),
+				ghttp.RespondWithJSONEncoded(http.StatusOK, expectedPV),
+			),
+		)
+	})
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	Context("Map Source Disks", func() {
+		It("looks up and applies PVC", func() {
+			vm := v1.VM{}
+
+			disk := v1.Disk{
+				Type: "PersistentVolumeClaim",
+				Source: v1.DiskSource{
+					Name: "test-claim",
+				},
+				Target: v1.DiskTarget{
+					Device: "vda",
+				},
+			}
+			disk.Type = "PersistentVolumeClaim"
+
+			domain := v1.DomainSpec{}
+			domain.Devices.Disks = []v1.Disk{disk}
+			vm.Spec.Domain = &domain
+
+			restClient := getRestClient(server.URL())
+			vmCopy, err := MapPersistentVolumes(&vm, restClient, kubeapi.NamespaceDefault)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(len(vmCopy.Spec.Domain.Devices.Disks)).To(Equal(1))
+			newDisk := vmCopy.Spec.Domain.Devices.Disks[0]
+			Expect(newDisk.Type).To(Equal("network"))
+			Expect(newDisk.Driver.Type).To(Equal("raw"))
+			Expect(newDisk.Driver.Name).To(Equal("qemu"))
+			Expect(newDisk.Device).To(Equal("disk"))
+			Expect(newDisk.Source.Protocol).To(Equal("iscsi"))
+			Expect(newDisk.Source.Name).To(Equal("iqn.2009-02.com.test:for.all/1"))
+		})
+	})
+})
+
+func getRestClient(url string) *rest.RESTClient {
```

Comment:
Instead of that function you can use `kubecli.Get(server.URL(), "").CoreV1().RESTCLient()